### PR TITLE
Add devtools panel open count histograms

### PR DESCRIPTION
**Pull request by tdsmith**
Hello! Devtools dashboarding relies heavily on longitudinal so I'd like to start migrating away given the deprecation timeline Frank announced.

I have a vague and possibly incorrect understanding that clients_daily is becoming the preferred derived dataset for dashboarding; is that right-ish? The docs explain that using main_summary from redash is discouraged; do you expect that using clients_daily will be more performant?

So I think the things I need to do are:

* Add devtools open count histogram sums to main_summary
* Add sums of open count histogram sums to clients_daily

Does this look like the right approach?

Thanks!

cc @fbertsch

### CODE REVIEW

**Comment by fbertsch** [comment]
@tdsmith huge props on taking this on yourself. What are your thoughts on just including those probes in the `histogramsWhitelist`? That's the more common layout of histograms these days.

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 1]
```diff
@@ -656,7 +656,38 @@ object MainSummaryView {
         (settings \ "defaultSearchEngineData" \ "origin").extractOpt[String],
         (settings \ "defaultSearchEngineData" \ "submissionURL").extractOpt[String],
         (settings \ "defaultSearchEngine").extractOpt[String],
+        hsum(histograms("parent") \ "DEVTOOLS_ABOUTDEBUGGING_OPENED_COUNT"),
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: If we include the histograms here, they also need to be included down below in the schema, right after here: https://github.com/mozilla/telemetry-batch-view/blob/master/src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala#L1050

Comment (fbertsch) [reply src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: However, if we just add them to the whitelist, it should work "as-is"

**Comment by tdsmith** [comment]
Thanks for taking a look! I started with the histogramsWhitelist approach but then got spooked that I didn't see any code in ClientsDailyView for summing sums-of-histograms. These are all count-type histograms which are morally scalars anyway, so there's not any value to maintaining their histogram nature in main_summary and having a scalar in main_summary is easier to work with.

I don't have strong feelings about switching to the whitelist approach but I'd need a hand with the right way to incant the clients_daily aggregation.

**Comment by fbertsch** [comment]
@tdsmith okay, makes sense. I recommend adding them to the whitelist first, and then we can sum them using e.g.
```
private def aggSum(field: String, alias: String): Column = sum(field).alias(alias)

aggSum('histogram_parent_devtools_aboutdebugging_opened_count[2]', 'devtools_aboutdebugging_opened_count')
```

We need a new function because we don't really want the name of the new column to be `histogram_parent_devtools_aboutdebugging_opened_count[2]_sum`. We would also want to test this.

If you're up for writing the new code and tests, go for it. Otherwise add the histograms and we'll do the rest on our end.

**Comment by tdsmith** [comment]
@fbertsch I took a stab at it; It happened that the two-argument `aggSum` was already implemented.

I don't know what to do with the error message that the test is throwing now:
```
[error] /telemetry-batch-view/src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestPayloads.scala:6:8: Could not write class com/mozilla/telemetry/views/ClientsDailyViewTestPayloads$ because it exceeds JVM code size limits. Method scala/Some's code too large!
[error] object ClientsDailyViewTestPayloads {
[error]        ^
[warn] 91 warnings found
[error] one error found
[error] (Test / compileIncremental) Compilation failed
```

**Comment by tdsmith** [comment]
@sunahsuh helped me poke at this a little; one finding was that defining `genericTests` with `def` instead of `val` changed the error message from `Method scala/Some's code too large!` to `Method genericTests’s code too large!`

@relud, Sunah said you might have some thoughts about how to tackle this; do you see a solution?

**Comment by relud** [comment]
I do see a solution. i think it's probably outside the scope of this PR. shouldn't take me too long to throw it together though. i'll cc you on a PR in a little bit.

**Comment by tdsmith** [comment]
Beautiful green! Thanks @relud and @sunahsuh for your help.

I tossed @relud's commit from #471 in here in order to test against it but I'll rebase it out if/when this looks ready to go.

**Comment by tdsmith** [comment]
This should be ready for review; cc @fbertsch

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 2]
```diff
@@ -955,6 +955,7 @@ object MainSummaryView {
       case _: BooleanHistogram => true
       case _: FlagHistogram => true // FIXME: hack to allow `A11Y_INSTANTIATED_FLAG` through until we can update it
       case _: CategoricalHistogram => true
+      case _: CountHistogram => true
```
[the remark this answers is not in the document]
Comment (tdsmith) [reply src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: That sounds much nicer; sounds good and thank you!

[the remark this answers is not in the document]
Comment (fbertsch) [reply src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: @tdsmith the PR is open at https://github.com/mozilla/telemetry-batch-view/pull/472, feel free to take a look!

**Comment by fbertsch** [comment]
@tdsmith okay, you should rebase now. This should simplify things!

**Comment by tdsmith** [comment]
Okay, this is r4r again!

**Pull request merged**